### PR TITLE
Proposal: MEP, the minikube enhancement proposal workflow

### PR DESCRIPTION
**Pull request by tstromberg**
A first draft at a suggested design proposal workflow for minikube. Based on KEP, as well as discussion from the minikube office hours.

The intent is to provide an early-feedback mechanism for roadmap-level implementation details.  

Comments and questions welcome!

### CODE REVIEW

**Comment by medyagh** [comment]
+1

**Comment by tstromberg** [comment]
Discussed at meeting and this seemed to be alright with folks. Can has approvals?

File: enhancements/proposed/20190925-minikube-enhancement-process.md [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+## Goals
+
+* Facilitate communication about the "how" and "why" of an enhancement before code is written
+* Lightweight enough to not deter casual contributions
+* A process applicable to any roadmap-worthy enhancement
+
+## Non-Goals
+
+* Coverage for smaller enhancements that would not be represented within the minikube roadmap.
+* Unnecessarily reductions in development velocity
```
Comment (woodcockjosh) [inline enhancements/proposed/20190925-minikube-enhancement-process.md]: grammar?

Comment (woodcockjosh) [inline enhancements/proposed/20190925-minikube-enhancement-process.md]: Grammar?

File: enhancements/proposed/20190925-minikube-enhancement-process.md [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+## Design Details
+
+The *minikube enhancement process (MEP)* is a way to proposed, communicate, and coordinate on new efforts for the minikube project. MEP is based on a simplification of the [Kubernetes Enhancement Process](https://github.com/kubernetes/enhancements/blob/master/keps/0001-kubernetes-enhancement-proposal-process.md).
```
Comment (woodcockjosh) [inline enhancements/proposed/20190925-minikube-enhancement-process.md]: propose*

File: enhancements/proposed/20190925-minikube-enhancement-process.md [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+
+### Pro's
+
+* Easier editing for trivial proposals
+
+### Con's
+
+* Authors may waste unneccessary time styling output
+* Styling may be inconsistent between proposals
+* Invisible to casual contributors to a project, as these proposals do not show up within the GitHub project page
```
Comment (woodcockjosh) [inline enhancements/proposed/20190925-minikube-enhancement-process.md]: Maybe approved proposals can be added to the GitHub projects page?

File: enhancements/proposed/20190925-minikube-enhancement-process.md (hunk 2 above)
Comment (josedonizetti) [inline enhancements/proposed/20190925-minikube-enhancement-process.md]:
```suggestion
The *minikube enhancement process (MEP)* is a way to propose, communicate, and coordinate on new efforts for the minikube project. MEP is based on a simplification of the [Kubernetes Enhancement Process](https://github.com/kubernetes/enhancements/blob/master/keps/0001-kubernetes-enhancement-proposal-process.md).
```

File: enhancements/README.md [hunk 4]
```diff
@@ -0,0 +1,13 @@
+# Enhancements
+
+The *minikube enhancement process (MEP)* is a way to proposed, communicate, and coordinate on new efforts for the minikube project. You can read the full details in the (MEP proposal)[proposed/20190925-minikube-enhancement-process.md]
```
Comment (josedonizetti) [inline enhancements/README.md]:
```suggestion
The *minikube enhancement process (MEP)* is a way to propose, communicate, and coordinate on new efforts for the minikube project. You can read the full details in the (MEP proposal)[proposed/20190925-minikube-enhancement-process.md]
```

File: enhancements/template.md [hunk 5]
```diff
@@ -0,0 +1,41 @@
+# Your proposal title
+
+First proposed: <date>
+Authors: <date>
```
Comment (woodcockjosh) [inline enhancements/template.md]: <full name>*

**Pull request merged**